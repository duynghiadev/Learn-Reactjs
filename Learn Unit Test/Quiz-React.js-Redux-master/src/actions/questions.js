// Node modules import
import axios from 'axios';

// Actions types import
import {
  FETCH_QUESTION,
  DELETE_CURRENT_QUESTION,
  INCREMENT_TOTAL_COUNT,
  INCREMENT_CORRECT_COUNT,
  RELOCATE_FROM_PROPOSITION_TO_BOARD,
  RELOCATE_FROM_BOARD_TO_PROPOSITION,
  CHECK_ANSWER,
  CHANGE_SCORE,
  SHOW_ERROR,
  HIDE_ERROR
 } from '../constants/questions.js';

// Functions import
import { stringToArray, shuffleArray } from '../helpers/functions.js'

// Receives a random question
export function fetchQuestion() {
  return function(dispatch) {
    return axios.get('http://jservice.io/api/random')
      .then(function(response) {
        // TODO: maybe delete this line
        const arrayOfCharacters = stringToArray(response.data[0].answer);
        const shuffledArray = shuffleArray(arrayOfCharacters);
        const data = {
          id: response.data[0].id,
          category: response.data[0].category,
          question: response.data[0].question,
          answer: response.data[0].answer,
          value: response.data[0].value,
          transformedAnswer: shuffledArray
        }
        dispatch({
          type: FETCH_QUESTION,
          payload: data
        })
      }, function(error) {
        dispatch({
          type: SHOW_ERROR
        })
      })
  }
}

// Hides an error and allowes to continue use the application
export function hideError() {
  return function(dispatch) {
    dispatch({
      type: HIDE_ERROR
    })
  }
}

// Delete current question (to see the loader)
export function removeQuestion() {
  return function(dispatch) {
    dispatch({
      type: DELETE_CURRENT_QUESTION
    })
  }
}

// Increments the total questions count
export function incrementTotalCount() {
  return function(dispatch) {
    dispatch({
      type: INCREMENT_TOTAL_COUNT
    })
  }
}

// Increments the correct questions count
export function incrementCorrectCount() {
  return function(dispatch) {
    dispatch({
      type: INCREMENT_CORRECT_COUNT
    })
  }
}

// Increments/decrements the score by the given amount
export function changeScore(amount) {
  return function(dispatch) {
    dispatch({
      type: CHANGE_SCORE,
      payload: amount
    })
  }
}

// Relocates character from  "charsInProposition" to "charsOnBoard"
export function relocateToBoard(char) {
  return function(dispatch) {
    dispatch({
      type: RELOCATE_FROM_PROPOSITION_TO_BOARD,
      payload: char
    })
  }
}

// Relocates character from  "charsOnBoard" to "charsInProposition"
export function relocateToProposition(char) {
  return function(dispatch) {
    dispatch({
      type: RELOCATE_FROM_BOARD_TO_PROPOSITION,
      payload: char
    })
  }
}

// Checks if the answer provided by user is correct
export function checkAnswer(char) {
  return function(dispatch) {
    dispatch({
      type: CHECK_ANSWER,
      payload: char
    })
  }
}
