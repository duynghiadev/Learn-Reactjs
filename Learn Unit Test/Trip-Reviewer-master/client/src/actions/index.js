import axios from 'axios';

import {
  GET_TRIPS,
  GET_USERS,
  GET_TRIP_W_REVIEWER,
  CLEAR_TRIP_W_REVIEWER,
  GET_TRIP_W_REVIEWER_ERROR,
  USER_REGISTER,
  USER_LOGIN,
  RESET_PASSWORD,
  USER_AUTH,
  ADD_TRIP,
  UPDATE_TRIP,
  UPDATE_USER,
  DELETE_TRIP,
  GET_TRIP,
  GET_TRIP_ERROR,
  CLEAR_NEW_TRIP,
  CLEAR_TRIP,
  GET_USER_REVIEWS,
} from '../constants/action-names';

export function getTrips(limit = 10, start = 0, order = 'asc', list = []) {
  const request = axios.get(`/api/getManyTrips?limit=${limit}&skip=${start}&order=${order}`);

  return dispatch =>
    request.then(response => {
      dispatch({
        type: GET_TRIPS,
        payload: {
          list: [...list, ...response.data],
          newTripsCount: response.data.length,
        },
      });
    });
}

export function getTripWithReviewer(id) {
  const request = axios.get(`/api/getTrip?id=${id}`);

  return dispatch =>
    request
      .then(({ data: tripData }) => {
        const trip = tripData;

        return axios.get(`/api/getReviewer?id=${trip.ownerId}`).then(({ data: reviewerData }) => {
          const response = {
            trip,
            reviewer: reviewerData,
          };

          dispatch({
            type: GET_TRIP_W_REVIEWER,
            payload: response,
          });
        });
      })
      .catch(error =>
        dispatch({
          type: GET_TRIP_W_REVIEWER_ERROR,
          payload: error,
        })
      );
}

export function getUserReviews(userId) {
  const request = axios.get(`/api/getUserReviews?user=${userId}`);

  return dispatch =>
    request.then(response => {
      dispatch({
        type: GET_USER_REVIEWS,
        payload: response.data,
      });
    });
}

export function updateUser(data) {
  const request = axios.post('/api/userUpdate', data);

  return dispatch =>
    request.then(response => {
      dispatch({
        type: UPDATE_USER,
        payload: response.data,
      });
    });
}

export function getTrip(id) {
  const request = axios.get(`/api/getTrip?id=${id}`);

  return dispatch =>
    request
      .then(response => {
        dispatch({
          type: GET_TRIP,
          payload: response.data,
        });
      })
      .catch(error => {
        dispatch({
          type: GET_TRIP_ERROR,
          payload: error,
        });
      });
}

export function updateTrip(data) {
  const request = axios.post('/api/tripUpdate', data);

  return dispatch =>
    request.then(response => {
      dispatch({
        type: UPDATE_TRIP,
        payload: response.data,
      });
    });
}

export function deleteTrip(id) {
  const request = axios.delete(`/api/tripDelete?id=${id}`);

  return dispatch =>
    request.then(response => {
      dispatch({
        type: DELETE_TRIP,
        payload: response.data,
      });
    });
}

export function clearTrip() {
  return {
    type: CLEAR_TRIP,
    payload: {
      trip: null,
      updatedTrip: false,
      postDeleted: false,
    },
  };
}

export function clearTripWithReviewer() {
  return {
    type: CLEAR_TRIP_W_REVIEWER,
    payload: {
      trip: null,
      reviewer: null,
    },
  };
}

export function loginUser({ email, password }) {
  const request = axios.post('/api/login', { email, password });

  return dispatch =>
    request.then(response => {
      dispatch({
        type: USER_LOGIN,
        payload: response.data,
      });
    });
}

export function resetPassword({ email }) {
  const request = axios.post('/api/resetPassword', { email });

  return dispatch =>
    request.then(response => {
      dispatch({
        type: RESET_PASSWORD,
        payload: response.data,
      });
    });
}

export function auth() {
  const request = axios.get('/api/auth');

  return dispatch =>
    request.then(response => {
      dispatch({
        type: USER_AUTH,
        payload: response.data,
      });
    });
}

export function addTrip(trip) {
  const request = axios.post('/api/trip', trip);

  return dispatch =>
    request.then(response => {
      dispatch({
        type: ADD_TRIP,
        payload: response.data,
      });
    });
}

export function clearNewTrip() {
  return {
    type: CLEAR_NEW_TRIP,
    payload: null,
  };
}

export function getUsers() {
  const request = axios.get('/api/users');

  return dispatch =>
    request.then(response => {
      dispatch({
        type: GET_USERS,
        payload: response.data,
      });
    });
}

export function userRegister(user) {
  const request = axios.post('/api/register', user);

  return dispatch =>
    request.then(({ data }) => {
      dispatch({
        type: USER_REGISTER,
        payload: data,
      });
    });
}
