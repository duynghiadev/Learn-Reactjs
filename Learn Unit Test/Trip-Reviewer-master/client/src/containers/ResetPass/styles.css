.reset-password-container {
  text-align: center;
  padding: 25px 0 50px 0;
}
.reset-password-container .form-element {
  margin-top: 20px;
}
.reset-password-container h2 {
  font-size: 33px;
  font-weight: 300;
  color: #02bcd4;
}
.reset-password-container input {
  font-size: 20px;
  border-radius: 3px;
  border: 1px solid #ebebeb;
  box-sizing: border-box;
  padding: 11px 31px;
  background: #f7f7f7;
  font-weight: 300;
  width: 260px;
}
.reset-password-container button {
  font-size: 20px;
  padding: 10px 52px;
  background: #02bcd4;
  color: #fff;
  font-weight: 300;
  border-radius: 3px;
  border: none;
  margin-top: 20px;
  width: 260px;
}
.reset-password-container .error {
  color: #ff5722;
  margin-top: 20px;
  font-weight: 600;
}
