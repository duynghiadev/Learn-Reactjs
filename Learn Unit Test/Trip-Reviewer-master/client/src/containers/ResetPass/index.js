import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { resetPassword } from '../../actions';

import './styles.css';

class ResetPass extends Component {
  state = {
    email: '',
  };

  handleInputEmail = event => {
    this.setState({ email: event.target.value });
  };

  submitForm = event => {
    event.preventDefault();
    this.props.dispatch(resetPassword(this.state));
  };

  render() {
    const { users } = this.props;

    return (
      <div className="reset-password-container">
        <form onSubmit={this.submitForm}>
          <h2>Reset Password</h2>

          <div className="form-element">
            <input
              type="email"
              placeholder="Enter your mail"
              value={this.state.email}
              onChange={this.handleInputEmail}
            />
          </div>

          <button type="submit">Reset</button>

          <div className="error">{users.reset ? <div>{users.reset.message}</div> : null}</div>
        </form>
      </div>
    );
  }
}

ResetPass.propTypes = {
  users: PropTypes.object.isRequired,
  dispatch: PropTypes.func.isRequired,
};

function mapStateToProps(state) {
  return {
    users: state.users,
  };
}

export default connect(mapStateToProps)(ResetPass);
