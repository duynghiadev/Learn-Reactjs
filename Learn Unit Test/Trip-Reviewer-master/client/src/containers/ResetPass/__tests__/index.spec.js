import React from 'react';
import configureStore from 'redux-mock-store';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';

import ResetPass from '..';
import { resetPassword } from '../../../actions';

const mockStore = configureStore();

jest.mock('./../../../actions', () => ({
  resetPassword: jest.fn(() => ({
    type: 'resetPassword',
  })),
}));

const mockComponent = (initialState = {}, props) => {
  const store = mockStore(initialState);

  return <ResetPass {...props} store={store} />;
};

describe('<ResetPass />', () => {
  it('should render component', () => {
    const initialState = {
      users: {},
    };
    const tree = create(mockComponent(initialState)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component with error message', () => {
    const initialState = {
      users: {
        reset: {
          message: 'test',
        },
      },
    };
    const tree = create(mockComponent(initialState)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should change state when handleInputEmail is called', () => {
    const initialState = {
      users: {},
    };
    const event = {
      target: {
        value: 'test',
      },
    };

    const instance = shallow(mockComponent(initialState)).dive().instance();

    instance.handleInputEmail(event);

    expect(instance.state.email).toEqual('test');
  });

  it('should dispatch resetPassword when submitForm is called', () => {
    const preventDefault = jest.fn();
    const state = {
      email: 'test',
    };
    const initialState = {
      users: {},
    };
    const event = { preventDefault };

    const instance = shallow(mockComponent(initialState)).dive().instance();
    instance.setState(state);

    instance.submitForm(event);

    expect(preventDefault).toHaveBeenCalled();
    expect(resetPassword).toHaveBeenCalledWith(state);
  });
});
