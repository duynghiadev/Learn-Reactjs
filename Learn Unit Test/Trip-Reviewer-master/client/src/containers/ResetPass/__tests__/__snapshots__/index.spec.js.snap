// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResetPass /> should render component 1`] = `
<div
  className="reset-password-container"
>
  <form
    onSubmit={[Function]}
  >
    <h2>
      Reset Password
    </h2>
    <div
      className="form-element"
    >
      <input
        onChange={[Function]}
        placeholder="Enter your mail"
        type="email"
        value=""
      />
    </div>
    <button
      type="submit"
    >
      Reset
    </button>
    <div
      className="error"
    />
  </form>
</div>
`;

exports[`<ResetPass /> should render component with error message 1`] = `
<div
  className="reset-password-container"
>
  <form
    onSubmit={[Function]}
  >
    <h2>
      Reset Password
    </h2>
    <div
      className="form-element"
    >
      <input
        onChange={[Function]}
        placeholder="Enter your mail"
        type="email"
        value=""
      />
    </div>
    <button
      type="submit"
    >
      Reset
    </button>
    <div
      className="error"
    >
      <div>
        test
      </div>
    </div>
  </form>
</div>
`;
